docs/developer.md

```diff
@@ -1,3 +1,51 @@
 # blob-uploader.js
 
-Coming soon...
+This library wraps the blob uploader service [Blob uploader](https://github.com/blinkmobile/blob-uploader) and AWS S3 to allow uploading and retrieval of blobs from S3 without having to work with AWS credentials and the AWS SDK directly.
+
+## Usage
+1.  Initialise the BlobUploader with the URL of the blob uploader service, e.g. 
+```
+new BlobUploader('https://bm-blob-uploader-dev.api.blinkm.io/')
+```
+2.  For simple upload via a single HTTP PUT, blobUploader.uploadBlob with the blob to be uploaded. This will return a Promise that will resolve with the id that can be used later to retrieve the blob.
+```
+blobUploader.uploadBlob(blob)
+  .then((id) => //store id or use to retrieve a GET URL as per step 3 below)
+```
+OR
+
+2.  For a managed upload which supports a progress event, cancellation and will adjust to network conditions, call blobUploader.managedUpload. This function takes as parameters the blob to be uploaded and a function that will be called with a progress event(OPTIONAL). This will return a Promise that will resolve with an object that includes the id that can be used later to retrieve the blob, a function upload()which returns a promise that will resolve when the upload succeeds or rejects if the upload fails or is stopped, and a function cancel() that will abort the upload. The progress event call will include a progress parameter with properties loaded and total (see [AWS SDK](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3/ManagedUpload.html#httpUploadProgress-event))
+```
+function progressFn (evt) {
+  const percentage = parseInt((evt.loaded * 100) / evt.total)
```

Comment:
Do you think its worth us passing `uploaded` and `total` as arguments instead of the `event` object?